java/cdk/src/main/java/sleeper/cdk/stack/bulkimport/EmrServerlessBulkImportStack.java

```diff
@@ -0,0 +1,245 @@
+        instanceProperties.getTags().forEach(
+                (key, value) -> tagKeyCondition.put("elasticmapreduce:RequestTag/" + key, value));
+
+        conditions.put("StringEquals", tagKeyCondition);
+
+        bulkImportJobStarter.addToRolePolicy(PolicyStatement.Builder.create()
+                .actions(List.of("elasticmapreduce:RunJobFlow")).effect(Effect.ALLOW)
+                .resources(List.of("*")).conditions(conditions).build());
+    }
+
+    public void createEmrServerlessApplication(InstanceProperties instanceProperties) {
+        String instanceId = instanceProperties.get(ID);
+        String region = instanceProperties.get(REGION);
+        String accountId = instanceProperties.get(ACCOUNT);
+        String repo = instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_CUSTOM_IMAGE_REPO);
+        String version = instanceProperties.get(VERSION);
+        String uri = accountId + ".dkr.ecr." + region + ".amazonaws.com/" + repo + ":" + version;
+
+        CfnApplicationProps props = CfnApplicationProps.builder()
+                .name(String.join("-", "sleeper", instanceId, "emr", "serverless"))
+                .releaseLabel(instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_RELEASE))
+                .architecture(instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_ARCHITECTURE))
+                .type("Spark")
+                .imageConfiguration(ImageConfigurationInputProperty.builder().imageUri(uri).build())
+                .networkConfiguration(NetworkConfigurationProperty.builder()
+                        .subnetIds(List.of(getSubnet(instanceProperties)))
+                        .securityGroupIds(List.of(createSecurityGroup(instanceProperties))).build())
+                .tags(List.of(new CfnTag.Builder().key("DeploymentStack")
+                        .value("EmrServerlessBulkImport").build()))
+                .build();
+
+        CfnApplication emrServerlessCluster = new CfnApplication(this, getArtifactId(), props);
+        instanceProperties.set(BULK_IMPORT_EMR_SERVERLESS_CLUSTER_NAME,
+                emrServerlessCluster.getName());
+        instanceProperties.set(BULK_IMPORT_EMR_SERVERLESS_APPLICATION_ID,
+                emrServerlessCluster.getAttrApplicationId());
+    }
+
+    private String getSubnet(InstanceProperties instanceProperties) {
+        List<String> subnets = instanceProperties.getList(SUBNETS);
+        return subnets.get(0);
+    }
+
+    private String createSecurityGroup(InstanceProperties instanceProperties) {
+        String instanceId = instanceProperties.get(ID);
+        IVpc vpc = Vpc.fromLookup(this, "VPC",
+                VpcLookupOptions.builder().vpcId(instanceProperties.get(VPC_ID)).build());
+
+        SecurityGroup securityGroup = SecurityGroup.Builder
+                .create(this, String.join("-", "SG", "sleeper", instanceId, "EMR-Serverless"))
+                .description("Security Group used by EMR Serverless").vpc(vpc).build();
```

Comment:
It's not necessary to set the instance ID in the logical CloudFormation ID here, since it will be nested under the stack for the instance. We can just set a simple ID like "EmrServerless".

Reply:
Updated to "EmrServerless"